servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/PublisherFlatMapSingle.java

```diff
@@ -180,7 +179,7 @@ public void onNext(T t) {
             // If we are cancelled the activeUpdater count is best effort depending upon which sources finish. This best
             // effort behavior mimics the semantics of cancel though so we don't take any special action to try to
             // adjust the count or prematurely terminate.
-            activeUpdater.incrementAndGet(this);
+            activeMappedSourcesUpdater.incrementAndGet(this);
```

Comment:
Since we are setting a negative value to this variable; should we instead use:
```java
activeMappedSourcesUpdater.accumulateAndGet(this, 1, (prev, next) -> prev >= 0 ? prev + next : prev);
```

Reply:
good question. `activeMappedSources` is set to a negative value in this Subscriber's `onComplete` method, so `onNext` should not see a negative value and I'll add an assert to clarify expectations.